### PR TITLE
Stories/select 2 stimulus

### PR DESCRIPTION
**Pull request by eanders**

### CODE REVIEW

**Comment by kwiliarty** [comment]
Before looking at the code I'm just testing the functionality in a browser. Some observations:

* External 'select all' is not reachable from the keyboard. It could be a button or a link rather than a div, or you could set `tabindex=0` and set a `role` or an `aria-role`
* On select #2 (and maybe others) if I select some items individually, then the select-none button does not work. This may be similar to the problem we saw with the sub-sections. Maybe we just accept it, but just wanted to point it out.
* Point of interest and maybe helpful for debugging: on the element that preselects multiple from an ajax load, the initial select none works. After that it does not.
* There is the known issue where the internal select-none does not clear the chips from the summary area.

Otherwise, it behaves as expected.

File: app/assets/javascripts/init.js.coffee [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #= require namespace
 App.init = ->
-  App.select2.init()
-  App.remoteSelectLoad.init()
+  # App.select2.init()
+  # App.remoteSelectLoad.init()
```
Comment (kwiliarty) [inline app/assets/javascripts/init.js.coffee]: Can we remove those lines now?

File: app/assets/javascripts/remote_select_load.js.coffee [hunk 2]
```diff
@@ -10,11 +10,11 @@ App.remoteSelectLoad.init = (root) =>
     loading_placeholder = 'Loading...'
     $select.attr('placeholder', loading_placeholder)
     if $select.hasClass('select2')
-      $select.select2('destroy')
+      # $select.select2('destroy')
```
Comment (kwiliarty) [inline app/assets/javascripts/remote_select_load.js.coffee]: Can we remove the commented lines?

**Comment by kwiliarty** [comment]
The select-all/none links are working well from the keyboard now. I did notice another minor keyboard accessibility issue with the multiselect dropdowns. I can use the return key to select items but not to unselect them. This is a lower priority because I can use backspacing in the chip area to remove items, but I thought I should mention it in case it is an easy fix.

**Comment by kwiliarty** [comment]
I think we should remove the obsolete select 2 examples from the "Form Elements" page in the styleguide: `/style_guide/form`

File: app/inputs/grouped_select_two_input.rb [hunk 3]
```diff
@@ -6,6 +6,29 @@
...
+
+  def input(wrapper_options = nil)
+    label_method, value_method = detect_collection_methods
+
+    options = input_html_options
+    options[:data] ||= {}
+    options[:data]['stimulus-select-target'] ||= ''
+    options[:data]['stimulus-select-target'] << ' element '
+    # options[:data][:action] ||= ''
+    # options[:data][:action] << ' change->stimulus-select#updateSelectAllStatus '
```
Comment (kwiliarty) [inline app/inputs/grouped_select_two_input.rb]: Remove commented code

File: app/inputs/select_two_input.rb [hunk 4]
```diff
@@ -6,6 +6,27 @@
...
+
+  def input(wrapper_options = nil)
+    label_method, value_method = detect_collection_methods
+
+    options = input_html_options
+    options[:data] ||= {}
+    options[:data]['stimulus-select-target'] ||= ''
+    options[:data]['stimulus-select-target'] << ' element '
+    # options[:data][:action] ||= ''
+    # options[:data][:action] << ' change->stimulus-select#updateSelectAllStatus '
```
Comment (kwiliarty) [inline app/inputs/select_two_input.rb]: Remove commented code

**Review by kwiliarty — COMMENTED** [review]
At this point it seems to me that the select-all and select-none buttons are not working for the dependent selects. Is that still in progress?

**Comment by eanders** [comment]
@kwiliarty , the select all/none links and internal items are working for me on the dependent selects.  Can you confirm that's still broken for you?  Also, all other items should be taken care of now.
One TODO:

- [ ] Search the codebase for `select2` to confirm anywhere we were adding the class is now listed as `as: :select2`
- [ ] Role editor needs re-implementation of select2

**Comment by kwiliarty** [comment]
The problem with the internal items on dependent selects seems to be sorted out. I started trying to update some of the old select2 calls to `as: :select_two` but in the process I found another issue with a general multi-select. If you select all and then start to delete chips you'll see that each click toggles the drop-down section. The chips are correctly removed, but it's disorienting to have the drop-down opening and closing. I think the correct behavior would be for the drop-down to stay open.

**Comment by kwiliarty** [comment]
Actually, it would also be fine for the drop-down to stay closed the whole time because the user can also see that the chip is removed.

**Comment by kwiliarty** [comment]
There's another complication when updating the existing select2's: Many of them are not using `simple_form` inputs, but the rails form helper `select_tag` with a `select2` class. I tried for a while to find a way to tell these to render as our custom input, but I could not. I might be missing something easy, of course. Otherwise, do we want/need to convert these to use simple form?

An example is in the view:

`drivers/client_access_control/app/views/client_access_control/clients/search/_filter.haml`

which displays at `/clients`

A tangential question that came up while I was working on this interface. Does this file ever get called? 

`app/views/clients/_filter.haml`

I feel like it might have been replaced by the views in the driver.

**Pull request merged**